core/src/databases/remote_databases/get_remote_database.rs

```diff
@@ -0,0 +1,21 @@
+use anyhow::Result;
+
+use crate::{
+    databases::remote_databases::remote_database::RemoteDatabase,
+    oauth::{client::OauthClient, credential::CredentialProvider},
+};
+
+use super::snowflake::SnowflakeRemoteDatabase;
+
+pub async fn get_remote_database(
+    credential_id: &str,
+) -> Result<Box<dyn RemoteDatabase + Sync + Send>> {
```

Comment:
needs to be `Box<dyn ...>` since we don't know the size of the underlying type (even if it is always Snowflake today).
needs to be `+ Sync + Send` otherwise we can't pass it to thread and axums rejects the handler.

TIL: you can use the [debug_handler](https://docs.rs/axum-macros/latest/axum_macros/attr.debug_handler.html) from `axum_macros` and it tells you exactly why you handler is rejected ! This is extremely useful as the rust compiler is clueless about those things.